File: app/src/androidTest/java/com/byoutline/kickmaterial/MainActivityFirstStartTest.java
package com.byoutline.kickmaterial;

import android.support.test.rule.ActivityTestRule;

import com.byoutline.kickmaterial.espressohelpers.DaggerRules;
import com.byoutline.kickmaterial.espressohelpers.WithCachedFieldIdlingResourcesTest;
import com.byoutline.kickmaterial.features.projectlist.MainActivity;

import org.junit.Rule;
import org.junit.Test;

import static com.byoutline.kickmaterial.espressohelpers.CommonEspressoCalls.viewWithTextIsDisplayed;

/**
 * @author Sebastian Kacprzak <sebastian.kacprzak at byoutline.com>
 */
public class MainActivityFirstStartTest extends WithCachedFieldIdlingResourcesTest {
    @Rule
    public ActivityTestRule<MainActivity> activityRule = DaggerRules.INSTANCE.userFirstLaunchRule();

    @Test
    public void testAllCategoriesShouldBeVisible() {
        viewWithTextIsDisplayed(R.string.all_categories);
    }

    @Test
    public void testHeaderShouldBeVisible() {
        viewWithTextIsDisplayed(R.string.explore);
    }
}

File: app/src/androidTest/java/com/byoutline/kickmaterial/MainActivityFirstStartTest.kt
package com.byoutline.kickmaterial

import com.byoutline.kickmaterial.espressohelpers.DaggerRules
import com.byoutline.kickmaterial.espressohelpers.viewWithTextIsDisplayed
import org.junit.Rule
import org.junit.Test

/**
 * @author Sebastian Kacprzak <sebastian.kacprzak at byoutline.com>
 */
class MainActivityFirstStartTest {
    @get:Rule
    val activityRule = DaggerRules.userFirstLaunchRule()
    @get:Rule
    val idlingResourceRule = DaggerRules.idlingResourceDiscoverField()

    @Test
    fun testAllCategoriesShouldBeVisible() {
        viewWithTextIsDisplayed(R.string.all_categories)
    }

    @Test
    fun testHeaderShouldBeVisible() {
        viewWithTextIsDisplayed(R.string.explore)
    }
}
